src/App.tsx

```diff
@@ -1,24 +1,286 @@
+
+      return;
+    }
+
+    const newTodo = {
+      id: 0,
+      userId: USER_ID,
+      title: query,
+      completed: false,
+    };
+
+    setTempTodo(newTodo);
+
+    try {
+      const createdTodo = await createTodo(newTodo);
+
+      setTodos((prevTodos) => [...prevTodos, createdTodo]);
+    } catch (error) {
+      setVisibleError(NewError.Add);
+    } finally {
+      setTempTodo(null);
+    }
+
+    setQuery('');
+  }, [query]);
+
+  const handleCloseError = useCallback(() => {
+    setVisibleError(null);
+  }, []);
+
+  const handleRemoveTodo = useCallback(async (todoId: number) => {
+    try {
+      await client.delete(`/todos/${todoId}`);
+
+      setTodos((prevTodos) => prevTodos.filter((todo) => todo.id !== todoId));
+    } catch (error) {
+      setVisibleError(NewError.Remove);
+    }
+  }, []);
+
+  const handleRemoveCompletedTodos = useCallback(async () => {
+    setIsRemovingCompleted(true);
+
+    try {
+      await Promise.all(
+        todos
+          .filter((todo) => todo.completed)
+          .map((todo) => deleteTodo(todo.id)),
+      );
+
+      setTodos((prevTodos) => prevTodos.filter((todo) => !todo.completed));
+    } catch (error) {
+      setVisibleError(NewError.Remove);
+    } finally {
+      setIsRemovingCompleted(false);
+    }
+  }, [todos]);
+
+  const handleUpdateTodo = useCallback(async (todoToUptade: Todo) => {
+    setUpdatingTodoId(todoToUptade.id);
+    const updatedTodo = {
+      ...todoToUptade,
+      completed: !todoToUptade.completed,
+    };
+
+    try {
+      await updateTodo(updatedTodo);
+
+      setTodos((prevTodos) => prevTodos.map((todo) => (
+        todo.id === todoToUptade.id ? updatedTodo : todo
+      )));
+    } catch (error) {
+      setVisibleError(NewError.Update);
+    } finally {
+      setUpdatingTodoId(null);
+    }
+  }, [todos]);
+
+  const handleUpdateAllTodos = useCallback(async () => {
+    setIsUpdatingEveryStatus(true);
+    try {
+      await Promise.all(
+        todos.map((todo) => updateTodo({
+          ...todo,
+          completed: !isEveryTotoCompleted,
+        })),
+      );
+
+      setTodos((prevTodos) => prevTodos.map((todo) => ({
+        ...todo,
+        completed: !isEveryTotoCompleted,
+      })));
+    } catch (error) {
+      setVisibleError(NewError.Update);
+    } finally {
+      setIsUpdatingEveryStatus(false);
+    }
+  }, [todos]);
+
+  const handleUpdateTodoTitle = useCallback(async (
+    todoToUpdate: Todo,
+    newTitle: string,
+  ) => {
+    setUpdatingTodoId(todoToUpdate.id);
+
+    try {
+      const updatedTodo = await updateTodo({
+        ...todoToUpdate,
+        title: newTitle,
+      });
+
+      setTodos((prevTodos) => prevTodos.map((todo) => (
+        todo.id === todoToUpdate.id ? updatedTodo : todo
+      )));
+    } catch (error) {
+      setVisibleError(NewError.Update);
+    } finally {
+      setUpdatingTodoId(null);
+    }
+  }, [todos]);
+
+  if (visibleError) {
+    setTimeout(() => {
```

Comment:
make it with useEffect || in Error component
Also, you should clearTimeout when component was deleted